src/renault_api/kamereon/models.py

```diff
@@ -351,9 +351,12 @@ def for_json(self) -> Dict[str, Any]:
             "activated": self.activated,
         }
         for day in helpers.DAYS_OF_WEEK:
-            schedule: Optional[ChargeDaySchedule] = self.__dict__.get(day)
-            if schedule:  # pragma: no branch
-                result[day] = schedule.for_json()
+            if day in vars(self):  # pragma: no branch
+                day_spec: Optional[ChargeDaySchedule] = getattr(self, day)
```

Comment:
What is the use of this condition? Is it not sufficient to use just getattr(self,day)? The attribute should always exist, even if it doesn't have a value.

```suggestion
            day_spec: Optional[ChargeDaySchedule] = getattr(self, day)
```